pkg/node/node.go

```diff
@@ -70,55 +70,45 @@ func NewNode(
 func (n *Node) AnnounceSignerPresence(
 	ctx context.Context,
 	operatorPublicKey *operator.PublicKey,
-	keepAddress common.Address,
-	keepMembersAddresses []common.Address,
+	keepID chain.ID,
+	keepMemberIDs []chain.ID,
 ) ([]tss.MemberID, error) {
-	broadcastChannel, err := n.networkProvider.BroadcastChannelFor(keepAddress.Hex())
+	broadcastChannel, err := n.networkProvider.BroadcastChannelFor(keepID.String())
 	if err != nil {
 		return nil, fmt.Errorf("failed to initialize broadcast channel: [%v]", err)
 	}
 
 	tss.RegisterUnmarshalers(broadcastChannel)
 
 	if err := broadcastChannel.SetFilter(
-		createAddressFilter(keepMembersAddresses),
+		createAddressFilter(keepMemberIDs),
 	); err != nil {
 		return nil, fmt.Errorf("failed to set broadcast channel filter: [%v]", err)
 	}
 
-	// TODO: Use generic type for representing address in node.go instead of
-	// common.Address from go-ethereum. The current implementation is not
-	// host-chain implementation agnostic.
-	// To do not propagate ethereum-specific types any further, we convert
-	// addresses to strings before passing them to AnnounceProtocol.
-	keepAddressString := keepAddress.Hex()
-	keepMembersAddressesStrings := make([]string, len(keepMembersAddresses))
-	for i, address := range keepMembersAddresses {
-		keepMembersAddressesStrings[i] = address.Hex()
-	}
 	return tss.AnnounceProtocol(
 		ctx,
 		operatorPublicKey,
-		keepAddressString,
-		keepMembersAddressesStrings,
+		keepID,
+		keepMemberIDs,
 		broadcastChannel,
-		n.ethereumChain.Signing().PublicKeyToAddress,
+		n.ethereumChain.PublicKeyToOperatorID,
 	)
 }
 
 func createAddressFilter(
-	addresses []common.Address,
+	keepMemberIDs []chain.ID,
 ) net.BroadcastChannelFilter {
-	authorizations := make(map[string]bool, len(addresses))
-	for _, address := range addresses {
-		authorizations[hex.EncodeToString(address.Bytes())] = true
+	operatorAuthorizations := make(map[string]bool, len(keepMemberIDs))
+	for _, keepMemberID := range keepMemberIDs {
+		operatorAuthorizations[keepMemberID.String()] = true
```

Comment:
This change breaks the key generation process. I observed a lot of logs like that:
```
keep-ecdsa rejecting message from [...]; author is not authorized
```
And the key is not generated as expected. I think the problem is because `keepMemberID.String()` represents an address hex with the checksum (according to `common.Address`'s `Hex` method). Down the road, in line `111`, we look at the authorization using an address constructed from encoded address bytes, without the checksum. This cause the mismatch and the autorization is not granted.

Lines 108-110 should be probably replaced by:
```
authorAddress := operator.PubkeyToAddress(*authorPublicKey).Hex()
```
But this cause a leak of `chain.ID` details as we need to know what `chain.ID`'s `String` method does under the hood.

Reply:
Fixed in https://github.com/keep-network/keep-ecdsa/pull/744/commits/9f6a0a6654ce61209c5b5c6967832ed7e3e25828. Some minor improvements has been also made in https://github.com/keep-network/keep-ecdsa/pull/744/commits/d24d3254e239d5772803e2e30bfe6050cca5bae8